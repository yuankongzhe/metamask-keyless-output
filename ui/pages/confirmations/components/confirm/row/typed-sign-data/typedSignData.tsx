import React from 'react';

import { sanitizeMessage } from '../../../../../../helpers/utils/util';
import { useI18nContext } from '../../../../../../hooks/useI18nContext';
import { Box } from '../../../../../../components/component-library';
import { BlockSize } from '../../../../../../helpers/constants/design-system';
import {
  ConfirmInfoRow,
  ConfirmInfoRowText,
} from '../../../../../../components/app/confirm/info/row';

import { DataTree } from './dataTree';

const parseMessage = (dataToParse: string) => {
  const { message, primaryType, types } = JSON.parse(dataToParse);
  const sanitizedMessage = sanitizeMessage(message, primaryType, types);
  return { sanitizedMessage, primaryType };
};

export const ConfirmInfoRowTypedSignData = ({ data }: { data: string }) => {
  const t = useI18nContext();

  if (!data) {
    return null;
  }

  const { sanitizedMessage, primaryType } = parseMessage(data);

  return (
    <Box width={BlockSize.Full} style={{ margin: '0 -8px' }}>
      <ConfirmInfoRow label={`${t('primaryType')}:`}>
        <ConfirmInfoRowText text={primaryType} />
      </ConfirmInfoRow>
      <Box style={{ margin: '0 -8px' }}>
        <DataTree data={sanitizedMessage.value} />
      </Box>
    </Box>
  );
};
