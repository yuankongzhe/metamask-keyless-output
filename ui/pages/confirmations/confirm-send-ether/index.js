export { default } from './confirm-send-ether';
