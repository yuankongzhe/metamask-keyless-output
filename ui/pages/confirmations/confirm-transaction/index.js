import ConfirmTransaction from './confirm-transaction.component';

export default ConfirmTransaction;
