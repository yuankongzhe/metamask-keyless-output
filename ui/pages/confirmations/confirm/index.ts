export { default as Confirm } from './confirm';
