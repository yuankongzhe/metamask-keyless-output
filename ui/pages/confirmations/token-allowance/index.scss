.token-allowance-container {
  &__icon-display-content {
    width: fit-content;
    height: 40px;
    box-sizing: border-box;
    border-radius: 100px;
    position: relative;

    &__siteimage-identicon {
      width: 24px;
      height: 24px;
      box-shadow: none;
      background: none;
    }
  }

  a.token-allowance-container__verify-link {
    width: fit-content;
    margin-inline-start: auto;
    margin-inline-end: auto;
    padding: 0;
  }

  a.token-allowance-container__view-details {
    width: fit-content;
    margin-inline-start: 108px;
    margin-inline-end: 108px;
  }

  &__card-wrapper {
    width: 100%;
  }

  &__data {
    width: 100%;
  }

  &__full-tx-content {
    max-width: 100%;
  }

  &__custom-nonce-warning {
    width: 100%;
    height: 30px;
  }
}
