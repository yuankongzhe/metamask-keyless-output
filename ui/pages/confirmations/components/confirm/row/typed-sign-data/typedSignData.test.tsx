import React from 'react';
import { render } from '@testing-library/react';

import { unapprovedTypedSignMsg } from '../../../../../../../test/data/confirmations/typed_sign';
import { ConfirmInfoRowTypedSignData } from './typedSignData';

describe('ConfirmInfoRowTypedSignData', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <ConfirmInfoRowTypedSignData
        data={unapprovedTypedSignMsg.msgParams.data}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should return null if data is not defined', () => {
    const { container } = render(<ConfirmInfoRowTypedSignData data={''} />);
    expect(container).toBeEmptyDOMElement();
  });
});
