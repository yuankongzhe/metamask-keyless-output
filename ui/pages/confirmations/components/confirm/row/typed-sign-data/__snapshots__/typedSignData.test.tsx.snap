// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmInfoRowTypedSignData should match snapshot 1`] = `
<div>
  <div
    class="mm-box mm-box--width-full"
    style="margin: 0px -8px;"
  >
    <div
      class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
      style="overflow-wrap: anywhere; min-height: 24px;"
    >
      <div
        class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-center"
      >
        <p
          class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
        >
          [primaryType]:
        </p>
      </div>
      <div
        class="mm-box mm-box--display-flex mm-box--gap-2 mm-box--flex-wrap-wrap mm-box--align-items-center"
      >
        <p
          class="mm-box mm-text mm-text--body-md mm-box--color-inherit"
        >
          Mail
        </p>
      </div>
    </div>
    <div
      class="mm-box"
      style="margin: 0px -8px;"
    >
      <div
        class="mm-box mm-box--margin-left-2"
      >
        <div
          class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
          style="overflow-wrap: anywhere; min-height: 24px;"
        >
          <div
            class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-center"
          >
            <p
              class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
            >
              Contents:
            </p>
          </div>
          <div
            class="mm-box mm-box--display-flex mm-box--gap-2 mm-box--flex-wrap-wrap mm-box--align-items-center"
          >
            <p
              class="mm-box mm-text mm-text--body-md mm-box--color-inherit"
            >
              Hello, Bob!
            </p>
          </div>
        </div>
        <div
          class="mm-box mm-box--margin-left-2"
        >
          <div
            class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
            style="overflow-wrap: anywhere; min-height: 24px;"
          >
            <div
              class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-center"
            >
              <p
                class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
              >
                Name:
              </p>
            </div>
            <div
              class="mm-box mm-box--display-flex mm-box--gap-2 mm-box--flex-wrap-wrap mm-box--align-items-center"
            >
              <p
                class="mm-box mm-text mm-text--body-md mm-box--color-inherit"
              >
                Cow
              </p>
            </div>
          </div>
          <div
            class="mm-box mm-box--margin-left-2"
          >
            <div
              class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
              style="overflow-wrap: anywhere; min-height: 24px;"
            >
              <div
                class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-center"
              >
                <p
                  class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
                >
                  0:
                </p>
              </div>
              <div
                class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--align-items-center"
              >
                <div
                  class="mm-box mm-text mm-avatar-base mm-avatar-base--size-xs mm-avatar-account mm-text--body-xs mm-text--text-transform-uppercase mm-box--display-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-text-default mm-box--background-color-background-alternative mm-box--rounded-full mm-box--border-color-transparent box--border-style-solid box--border-width-1"
                >
                  <div
                    class="mm-avatar-account__jazzicon"
                  >
                    <div
                      style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 16px; height: 16px; display: inline-block; background: rgb(35, 98, 225);"
                    >
                      <svg
                        height="16"
                        width="16"
                        x="0"
                        y="0"
                      >
                        <rect
                          fill="#FA7900"
                          height="16"
                          transform="translate(-0.1251869550000172 -0.18581992328639055) rotate(237.8 8 8)"
                          width="16"
                          x="0"
                          y="0"
                        />
                        <rect
                          fill="#034F5E"
                          height="16"
                          transform="translate(5.484089776243406 -5.806818790918918) rotate(321.2 8 8)"
                          width="16"
                          x="0"
                          y="0"
                        />
                        <rect
                          fill="#FB185C"
                          height="16"
                          transform="translate(4.337646399293827 10.83159092527544) rotate(190.8 8 8)"
                          width="16"
                          x="0"
                          y="0"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div>
                  <div
                    aria-describedby="tippy-tooltip-1"
                    class=""
                    data-original-title="0xCD2a3d9F938E13CD947Ec05AbC7FE734Df8DD826"
                    data-tooltipped=""
                    style="display: inline;"
                    tabindex="0"
                  >
                    <p
                      class="mm-box mm-text mm-text--body-md mm-box--margin-left-2 mm-box--color-inherit"
                    >
                      0xCD2a3...DD826
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
              style="overflow-wrap: anywhere; min-height: 24px;"
            >
              <div
                class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-center"
              >
                <p
                  class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
                >
                  1:
                </p>
              </div>
              <div
                class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--align-items-center"
              >
                <div
                  class="mm-box mm-text mm-avatar-base mm-avatar-base--size-xs mm-avatar-account mm-text--body-xs mm-text--text-transform-uppercase mm-box--display-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-text-default mm-box--background-color-background-alternative mm-box--rounded-full mm-box--border-color-transparent box--border-style-solid box--border-width-1"
                >
                  <div
                    class="mm-avatar-account__jazzicon"
                  >
                    <div
                      style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 16px; height: 16px; display: inline-block; background: rgb(24, 183, 242);"
                    >
                      <svg
                        height="16"
                        width="16"
                        x="0"
                        y="0"
                      >
                        <rect
                          fill="#F91E01"
                          height="16"
                          transform="translate(-3.031601448459508 1.44638157917772) rotate(179.5 8 8)"
                          width="16"
                          x="0"
                          y="0"
                        />
                        <rect
                          fill="#FA5300"
                          height="16"
                          transform="translate(0.001001577275992914 -7.592133809870717) rotate(370.7 8 8)"
                          width="16"
                          x="0"
                          y="0"
                        />
                        <rect
                          fill="#F5A300"
                          height="16"
                          transform="translate(-9.667744843805323 10.267175012988094) rotate(298.9 8 8)"
                          width="16"
                          x="0"
                          y="0"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div>
                  <div
                    aria-describedby="tippy-tooltip-2"
                    class=""
                    data-original-title="0xDeaDbeefdEAdbeefdEadbEEFdeadbeEFdEaDbeeF"
                    data-tooltipped=""
                    style="display: inline;"
                    tabindex="0"
                  >
                    <p
                      class="mm-box mm-text mm-text--body-md mm-box--margin-left-2 mm-box--color-inherit"
                    >
                      0xDeaDb...DbeeF
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
              style="overflow-wrap: anywhere; min-height: 24px;"
            >
              <div
                class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-center"
              >
                <p
                  class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
                >
                  2:
                </p>
              </div>
              <div
                class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--align-items-center"
              >
                <div
                  class="mm-box mm-text mm-avatar-base mm-avatar-base--size-xs mm-avatar-account mm-text--body-xs mm-text--text-transform-uppercase mm-box--display-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-text-default mm-box--background-color-background-alternative mm-box--rounded-full mm-box--border-color-transparent box--border-style-solid box--border-width-1"
                >
                  <div
                    class="mm-avatar-account__jazzicon"
                  >
                    <div
                      style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 16px; height: 16px; display: inline-block; background: rgb(245, 143, 0);"
                    >
                      <svg
                        height="16"
                        width="16"
                        x="0"
                        y="0"
                      >
                        <rect
                          fill="#FA3E00"
                          height="16"
                          transform="translate(2.1189727229825075 3.7201589705667972) rotate(163.6 8 8)"
                          width="16"
                          x="0"
                          y="0"
                        />
                        <rect
                          fill="#018E74"
                          height="16"
                          transform="translate(-3.5064720430845115 9.598091001429925) rotate(205.4 8 8)"
                          width="16"
                          x="0"
                          y="0"
                        />
                        <rect
                          fill="#238FE1"
                          height="16"
                          transform="translate(-5.001213401730692 12.04822844354251) rotate(286.5 8 8)"
                          width="16"
                          x="0"
                          y="0"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div>
                  <div
                    aria-describedby="tippy-tooltip-3"
                    class=""
                    data-original-title="0x06195827297c7A80a443b6894d3BDB8824b43896"
                    data-tooltipped=""
                    style="display: inline;"
                    tabindex="0"
                  >
                    <p
                      class="mm-box mm-text mm-text--body-md mm-box--margin-left-2 mm-box--color-inherit"
                    >
                      0x06195...43896
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="mm-box mm-box--margin-left-2"
        >
          <div
            class="mm-box mm-box--margin-left-2"
          >
            <div
              class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
              style="overflow-wrap: anywhere; min-height: 24px;"
            >
              <div
                class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-center"
              >
                <p
                  class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
                >
                  Name:
                </p>
              </div>
              <div
                class="mm-box mm-box--display-flex mm-box--gap-2 mm-box--flex-wrap-wrap mm-box--align-items-center"
              >
                <p
                  class="mm-box mm-text mm-text--body-md mm-box--color-inherit"
                >
                  Bob
                </p>
              </div>
            </div>
            <div
              class="mm-box mm-box--margin-left-2"
            >
              <div
                class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
                style="overflow-wrap: anywhere; min-height: 24px;"
              >
                <div
                  class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-center"
                >
                  <p
                    class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
                  >
                    0:
                  </p>
                </div>
                <div
                  class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--align-items-center"
                >
                  <div
                    class="mm-box mm-text mm-avatar-base mm-avatar-base--size-xs mm-avatar-account mm-text--body-xs mm-text--text-transform-uppercase mm-box--display-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-text-default mm-box--background-color-background-alternative mm-box--rounded-full mm-box--border-color-transparent box--border-style-solid box--border-width-1"
                  >
                    <div
                      class="mm-avatar-account__jazzicon"
                    >
                      <div
                        style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 16px; height: 16px; display: inline-block; background: rgb(242, 162, 2);"
                      >
                        <svg
                          height="16"
                          width="16"
                          x="0"
                          y="0"
                        >
                          <rect
                            fill="#C8144A"
                            height="16"
                            transform="translate(3.3716958087807476 -1.5355449411453455) rotate(359.3 8 8)"
                            width="16"
                            x="0"
                            y="0"
                          />
                          <rect
                            fill="#01878E"
                            height="16"
                            transform="translate(-4.56026532072257 -4.118767169186573) rotate(335.8 8 8)"
                            width="16"
                            x="0"
                            y="0"
                          />
                          <rect
                            fill="#1897F2"
                            height="16"
                            transform="translate(-12.66850392914138 2.191919563316492) rotate(275.0 8 8)"
                            width="16"
                            x="0"
                            y="0"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <div>
                    <div
                      aria-describedby="tippy-tooltip-4"
                      class=""
                      data-original-title="0xbBbBBBBbbBBBbbbBbbBbbbbBBbBbbbbBbBbbBBbB"
                      data-tooltipped=""
                      style="display: inline;"
                      tabindex="0"
                    >
                      <p
                        class="mm-box mm-text mm-text--body-md mm-box--margin-left-2 mm-box--color-inherit"
                      >
                        0xbBbBB...bBBbB
                      </p>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
                style="overflow-wrap: anywhere; min-height: 24px;"
              >
                <div
                  class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-center"
                >
                  <p
                    class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
                  >
                    1:
                  </p>
                </div>
                <div
                  class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--align-items-center"
                >
                  <div
                    class="mm-box mm-text mm-avatar-base mm-avatar-base--size-xs mm-avatar-account mm-text--body-xs mm-text--text-transform-uppercase mm-box--display-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-text-default mm-box--background-color-background-alternative mm-box--rounded-full mm-box--border-color-transparent box--border-style-solid box--border-width-1"
                  >
                    <div
                      class="mm-avatar-account__jazzicon"
                    >
                      <div
                        style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 16px; height: 16px; display: inline-block; background: rgb(250, 54, 0);"
                      >
                        <svg
                          height="16"
                          width="16"
                          x="0"
                          y="0"
                        >
                          <rect
                            fill="#2395E1"
                            height="16"
                            transform="translate(-1.7353143855136905 0.5115867848025443) rotate(191.9 8 8)"
                            width="16"
                            x="0"
                            y="0"
                          />
                          <rect
                            fill="#F90101"
                            height="16"
                            transform="translate(1.7169232764557802 -9.276715881533601) rotate(319.8 8 8)"
                            width="16"
                            x="0"
                            y="0"
                          />
                          <rect
                            fill="#F58700"
                            height="16"
                            transform="translate(9.91414735179574 5.139720751783702) rotate(180.7 8 8)"
                            width="16"
                            x="0"
                            y="0"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <div>
                    <div
                      aria-describedby="tippy-tooltip-5"
                      class=""
                      data-original-title="0xB0BdaBea57B0BDABeA57b0bdABEA57b0BDabEa57"
                      data-tooltipped=""
                      style="display: inline;"
                      tabindex="0"
                    >
                      <p
                        class="mm-box mm-text mm-text--body-md mm-box--margin-left-2 mm-box--color-inherit"
                      >
                        0xB0Bda...bEa57
                      </p>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
                style="overflow-wrap: anywhere; min-height: 24px;"
              >
                <div
                  class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-center"
                >
                  <p
                    class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
                  >
                    2:
                  </p>
                </div>
                <div
                  class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--align-items-center"
                >
                  <div
                    class="mm-box mm-text mm-avatar-base mm-avatar-base--size-xs mm-avatar-account mm-text--body-xs mm-text--text-transform-uppercase mm-box--display-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-text-default mm-box--background-color-background-alternative mm-box--rounded-full mm-box--border-color-transparent box--border-style-solid box--border-width-1"
                  >
                    <div
                      class="mm-avatar-account__jazzicon"
                    >
                      <div
                        style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 16px; height: 16px; display: inline-block; background: rgb(200, 20, 86);"
                      >
                        <svg
                          height="16"
                          width="16"
                          x="0"
                          y="0"
                        >
                          <rect
                            fill="#018E8C"
                            height="16"
                            transform="translate(3.988074950687866 0.3554940504644344) rotate(177.8 8 8)"
                            width="16"
                            x="0"
                            y="0"
                          />
                          <rect
                            fill="#F29202"
                            height="16"
                            transform="translate(-7.220928666254465 -2.4813555582770794) rotate(334.6 8 8)"
                            width="16"
                            x="0"
                            y="0"
                          />
                          <rect
                            fill="#236FE1"
                            height="16"
                            transform="translate(14.621768557060026 -6.282465882779628) rotate(456.1 8 8)"
                            width="16"
                            x="0"
                            y="0"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <div>
                    <div
                      aria-describedby="tippy-tooltip-6"
                      class=""
                      data-original-title="0xB0B0b0b0b0b0B000000000000000000000000000"
                      data-tooltipped=""
                      style="display: inline;"
                      tabindex="0"
                    >
                      <p
                        class="mm-box mm-text mm-text--body-md mm-box--margin-left-2 mm-box--color-inherit"
                      >
                        0xB0B0b...00000
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
